module-operator.tf

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  operator_private_ip = (var.create_cluster && var.create_operator
+    ? one(module.operator[*].private_ip)
+    : var.operator_private_ip
+  )
+
+  operator_enabled = alltrue([
+    (var.create_cluster || coalesce(var.cluster_id, "none") != "none"),
+    (var.create_operator || coalesce(var.operator_private_ip, "none") != "none"),
+  ])
+
+  operator_image_id = (var.operator_image_type == "custom"
+    ? var.operator_image_id : element(tolist(setintersection([
+      lookup(local.image_ids, format("%s %s", var.operator_image_os, split(".", var.operator_image_os_version)[0]), null),
+      local.image_ids.nongpu, local.image_ids.x86_64,
+    ]...)), 0)
+  )
+}
+
+module "operator" {
```

Comment:
As above for bastion. It's a reusable module.

Reply:
The operator in this context is a single Compute instance that is specialized for OKE with installed packages + kubeconfig, and highly coupled to other components with connected NSGs, subnet creation, input/output variables, consistent tag definition, and IAM policies. There's not enough here IMO to warrant the overhead of maintaining it outside of the project and keeping in sync. `5.x` is making it easier to apply a partial set of components, e.g. using existing VCN/subnets with cluster/NSG creation disabled - it'd be possible to apply the operator component on its own with an existing `cluster_id` this way, though I'd question why anyone should need to. I don't see any other references to the project.

Reply:
Discussed and in agreement for now - may revisit in a subsequent PR w/ enhancements to related projects.